components/src/core/components/Input/DropdownInput.vue

```diff
@@ -0,0 +1,390 @@
+            '--disabled': option.disabled,
+            '--selected': option.selected,
+            '--focused': isFocused(index),
+          }"
+          @mousedown="onSelectOption(option)"
+        >
+          {{ option.label }}
+        </li>
+        <li
+          v-if="filteredOptions.length === 0"
+          class="oxd-dropdown-options-item --disabled"
+        >
+          No options
+        </li>
+      </ul>
+    </transition>
+    <div class="oxd-dropdown-selected" v-if="multiple">
+      <oxd-chip
+        v-for="(option, index) in selected"
+        :key="`${index}-selected-${option.id}`"
+        :label="option.label"
+        class="oxd-dropdown-selected-chip"
+      >
+        <oxd-icon class="--clear" name="x" @click="onRemoveOption(option)" />
+      </oxd-chip>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Chip from '@orangehrm/oxd/core/components/Chip/Chip.vue';
+import Icon from '@orangehrm/oxd/core/components/Icon/Icon.vue';
+import Input from '@orangehrm/oxd/core/components/Input/Input.vue';
+import Spinner from '@orangehrm/oxd/core/components/Loader/Spinner.vue';
+import debounce from '../../../utils/debounce';
+import {
+  defineComponent,
+  onBeforeMount,
+  PropType,
+  reactive,
+  toRefs,
+  watch,
+} from 'vue';
+
+interface Option {
+  id: number;
+  label: string;
+  disabled: boolean;
+  selected: boolean;
+}
+
+export default defineComponent({
+  name: 'oxd-dropdown-input',
+
+  props: {
+    modelValue: {},
+    hasError: {
+      type: Boolean,
+      default: false,
+    },
+    multiple: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    preSelect: {
+      type: Boolean,
+      default: false,
+    },
+    clear: {
+      type: Boolean,
+      default: true,
+    },
+    lazyLoad: {
+      type: Boolean,
+      default: false,
+    },
+    disabledOptions: {
+      type: Array as PropType<number[]>,
+      default: [],
+    },
+    selectedOptions: {
+      type: Array as PropType<number[]>,
+      default: [],
+    },
+    placeholderText: {
+      type: String,
+      default: 'Type for hints...',
+    },
+    options: {
+      type: [Array, Function],
+      default: () => {
+        return [];
+      },
+    },
+  },
+
+  emits: [
+    'update:modelValue',
+    'blur',
+    'dropdown:cleared',
+    'dropdown:opened',
+    'dropdown:closed',
+  ],
+
+  components: {
+    'oxd-chip': Chip,
+    'oxd-icon': Icon,
+    'oxd-input': Input,
+    'oxd-loading-spinner': Spinner,
+  },
+
+  setup(props, {emit}) {
+    const state = reactive({
+      searchTerm: '',
+      open: false,
+      focused: false,
+      selected: [],
+      localOptions: [],
+      isLoading: false,
+      pointer: 0,
+    });
+
+    watch(
+      () => state.selected,
+      function() {
+        emit('update:modelValue', state.selected);
+      },
+    );
+
+    // Mark preselected options
+    const preselectOptions = () => {
```

Comment:
Why don't use computed props here?

Reply:
Here `on setup` props passed as `preSelected` will be put to `selected options` array. If we were to use a computed property for `selected options`, we will have to maintain another array since `selected options` are also added/removed by user interaction (ie: clicking).